### ISSUE TITLE
Allow drag and dropping audio effects on an audio bus

### ISSUE THREAD

**Issue by isaacremnant** [godotengine/godot#17066]
**Godot version:** 3

It would be convenient to be able to drag and drop a saved AudioEffect file on top of an audio bus to add the effect to it, see: 
![audioeffect](https://user-images.githubusercontent.com/35355161/36702001-5f271172-1b23-11e8-8db4-dbfaf9e0eda0.png)

I haven't found a way to add a saved effect to the bus layout using the editor, so drag and dropping would be the most intuitive way to add this functionality.

**Comment by akien-mga** [comment]
Makes sense. This should be fairly easy to implement, refer to other editor plugins that support drag and drop to see how it's implemented.

**Comment by Waranoi** [comment]
I took a shot at this here #42288. 

As mentioned in the PR, the effect name in the bus will be empty unless the user explicitly sets the name of the resource before drag and dropping it. I would have not realized that it's because the name of a newly created audio effect resource is empty by default unless I had my debugger available. The user will probably spend a lot of time before realizing why their effect doesn't have a name in their audio bus. Should there maybe be a separate PR to set the name of an audio effect resource to its filename when created? Also changing the name of an audio effect resource doesn't automatically update the bus layout, I don't know what the best solution for that would be.

![image](https://user-images.githubusercontent.com/31479779/94080260-06c0ae00-fdff-11ea-8c86-bc4a4f6da07c.png) ![image](https://user-images.githubusercontent.com/31479779/94080325-0b856200-fdff-11ea-9b8d-7f3e53094618.png)


In hindsight I should have posted this comment before issuing a PR, apologies, it's my first time contributing to a group project.
